src/main/java/org/apache/pirk/encryption/Paillier.java

```diff
@@ -340,4 +342,41 @@ private String parametersToString()
     return "p = " + p.intValue() + " q = " + q.intValue() + " N = " + N.intValue() + " NSquared = " + NSquared.intValue() + " lambdaN = " + lambdaN.intValue()
         + " bitLength = " + bitLength;
   }
+
+  @Override public boolean equals(Object o)
+  {
+    if (this == o)
+      return true;
+    if (o == null || getClass() != o.getClass())
+      return false;
+
+    Paillier paillier = (Paillier) o;
+
+    if (bitLength != paillier.bitLength)
+      return false;
+    if (!p.equals(paillier.p))
+      return false;
+    if (!q.equals(paillier.q))
+      return false;
+    if (!N.equals(paillier.N))
+      return false;
+    if (!NSquared.equals(paillier.NSquared))
+      return false;
+    if (!lambdaN.equals(paillier.lambdaN))
+      return false;
+    return w.equals(paillier.w);
+
+  }
+
+  @Override public int hashCode()
+  {
+    int result = p.hashCode();
+    result = 31 * result + q.hashCode();
+    result = 31 * result + N.hashCode();
+    result = 31 * result + NSquared.hashCode();
+    result = 31 * result + lambdaN.hashCode();
+    result = 31 * result + w.hashCode();
+    result = 31 * result + bitLength;
+    return result;
+  }
```

Comment:
Certainly one could. My IDE auto-generated these, and after that I just checked them over to make sure they made sense.

Honestly, Objects.hash really is kind of pretty. I'll change this over to use it for all the hashCode methods I generated for this code.

While I'm doing that, and since a quick google didn't seem to produce useful results (or it could be me; the hour is late), is Objects.hash going to Do The Right Thing when I feed it primitives?